### ISSUE TITLE
[FEATURE] export useOpenFeatureClientStatus

### ISSUE THREAD

**Issue by literalpie** [open-feature/js-sdk#1036]
### Requirements

the hook in use-open-feature-client-status.ts would be useful, but is not exported. I'm not sure if this is intentional or not?

**Comment by beeme1mr** [comment]
Hey @literalpie, would the `useWhenProviderReady` hook satisfy your use case?

**Comment by literalpie** [comment]
No, I'm checking for NOT_READY because I want to defer rendering of my feature flagged component until the loading state is finished (and I want to treat the error state fallback the same as ready)

It's simple enough to replicate the hook. I just opened the issue in case it's an oversight.

**Comment by toddbaert** [comment]
Maybe I'm confused, but what you're describing is the intended purposed of the `useWhenProviderReady` hook, though it has some extras (to support suspense and re-rendering); you can disable these if you don't want them though.

**Comment by literalpie** [comment]
useWhenProviderReady will return false if the status is ERROR or FATAL, or whatever. I want it to return true.

**Comment by toddbaert** [comment]
> useWhenProviderReady will return false if the status is ERROR or FATAL, or whatever. I want to treat that as true.

Ah, I see.

Ya in that case I can see the value. I think it would be nice to export this, but in that case I'd like it to have the behavior and features of or other public hooks (suspense, re-rendering, etc), like [this](https://github.com/open-feature/js-sdk/blob/main/packages/react/src/provider/use-when-provider-ready.ts#L21). Does that make sense to you @literalpie ? That way our hooks have consistent features and options.

I can update the issue, and I'm happy to assign it to you if you'd like to work on it (no pressure though).

**Comment by literalpie** [comment]
Sorry, I don't think I'll find time to work on this in the near term.

I'm also not sure I understand what you mean by "re-rendering" from other public hooks. Because this hook sets a "useState" state when the events happen, I think it will cause re-rendering as much as needed.

Also, because this hook is only getting the status of the client, and not doing any asynchronous work, I'm not sure suspense would be used. (but I rarely use suspense, so maybe I just don't understand right)
